src/Model/State.re

```diff
@@ -26,12 +26,13 @@ type t = {
   contextMenu: option(ContextMenu.t(Actions.t)),
   mode: Vim.Mode.t,
   completions: Completions.t,
+  configuration: Configuration.t,
+  decorationProviders: list(DecorationProvider.t),
```

Comment:
The decoration providers were moved out of SCM because it's more closely related to the FileExplorer. Once there's a feature project for that, this should probably be moved there.